src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala

```diff
@@ -239,26 +239,27 @@ private class TaskLauncherActor(
   }
 
   private[this] def receiveInstanceUpdate: Receive = {
-    case update: InstanceDeleted =>
-      logger.info(s"receiveInstanceUpdate: ${update.id} was deleted (${update.condition})")
-      // A) If the app has constraints, we need to reconsider offers that
-      // we already rejected. E.g. when a host:unique constraint prevented
-      // us to launch tasks on a particular node before, we need to reconsider offers
-      // of that node after a task on that node has died.
-      //
-      // B) If a reservation timed out, already rejected offers might become eligible for creating new reservations.
-      if (runSpec.constraints.nonEmpty || (runSpec.isResident && shouldLaunchInstances)) {
-        maybeOfferReviver.foreach(_.reviveOffers())
+    case update: InstanceUpdated =>
```

Comment:
This is the same as before, since we now keep the instance instead of expunging it and creating a new one. The handling of `InstanceDeleted` was moved to below.

**Before**: revive offers and sync instance when it has been deleted (== is is terminal).
**Now**: revive offers and sync instance when it has was updated and is now terminal and scheduled.

Reply:
(discussed in person)